frontend/src/pages/SignUp.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+
+function SignUp() {
+  const [formData, setFormData] = useState({
+    username: '',
+    email: '',
+    password: '',
+    confirm_password: ''
+  });
+  const [message, setMessage] = useState('');
+  const [errors, setErrors] = useState({});
+
+  const handleChange = (e) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    // Check if passwords match
+    if (formData.password !== formData.confirm_password) {
+      setErrors({ confirm_password: ["Passwords do not match"] });
+      return;
+    }
+
+    const response = await fetch('http://127.0.0.1:8000/api/register/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(formData)
+    });
+
+    const data = await response.json();
+
+    if (response.ok) {
+      setMessage('User created successfully');
+      setErrors({});
+    } else {
+      setErrors(data);
+    }
+  };
+
+  return (
+    <div className="container">
+      <h2>Sign Up</h2>
+      <form onSubmit={handleSubmit}>
+        <div className="form-group">
+          <label>Username</label>
+          <input
```

Comment:
Create a constant config object for all input fields. Iterate over it and render each field.